com.incquerylabs.v4md/src/main/com/incquerylabs/v4md/ViatraQueryAdapter.java

```diff
@@ -144,6 +148,7 @@ public void executeActionOnEngine(Consumer<AdvancedViatraQueryEngine> action) {
 					return false;
 				});
 			if(!isInitialized) {
+				engine.ifPresent(AdvancedViatraQueryEngine::dispose);
```

Comment:
It would have been nice for this update to be mentioned in the commit comment as well. It makes sense to have it here, but this seems unrelated to the fix itself.

I know this is not really possible to fix in Github, but in Gerrit I would have asked to do this in a separate change/commit.